controller/cmd/public-api/main.go

```diff
@@ -41,25 +41,40 @@ func main() {
 	if err != nil {
 		log.Fatal(err.Error())
 	}
-	spClient, err := k8s.NewSpClientSet(*kubeConfigPath)
-	if err != nil {
-		log.Fatal(err.Error())
-	}
+
+	var k8sAPI *k8s.API
 	restrictToNamespace := ""
 	if *singleNamespace {
 		restrictToNamespace = *controllerNamespace
+
+		k8sAPI = k8s.NewAPI(
+			k8sClient,
+			nil,
+			restrictToNamespace,
+			k8s.Deploy,
+			k8s.Pod,
+			k8s.RC,
+			k8s.RS,
+			k8s.Svc,
+		)
+	} else {
```

Comment:
i realize this pattern was already in place, and exists in proxy-api, but consider a small DRY up like this:
```go
var spClient *spclient.Clientset
restrictToNamespace := ""
resources := []k8s.APIResource{k8s.Deploy, k8s.Pod, k8s.RC, k8s.RS, k8s.Svc}

if *singleNamespace {
	restrictToNamespace = *controllerNamespace
} else {
	spClient, err = k8s.NewSpClientSet(*kubeConfigPath)
	if err != nil {
		log.Fatal(err.Error())
	}

	resources = append(resources, k8s.SP)
}

k8sAPI := k8s.NewAPI(
	k8sClient,
	spClient,
	restrictToNamespace,
	resources...,
)
```

Reply:
If you do go this route, mind updating controller/command/proxy-api/main.go to match?